src/nspawn/nspawn-cgroup.c

```diff
@@ -100,14 +111,23 @@ int sync_cgroup(pid_t pid, CGroupUnified unified_requested) {
 
         undo_mount = true;
 
+        fn = strjoina(tree, cgroup);
+        (void) rm_rf(fn, REMOVE_ROOT|REMOVE_ONLY_DIRECTORIES);
```

Comment:
I cannot reproduce this. With the first recipe, things work just fine (without the rm_rf part). At least with the host running unified. I'll reboot and test with the host running mixed later on.

With the second recipe, the container survives the death of nspawn. But in this case we have a mess, so it's probably if things fail.